### PR TITLE
Provide the option to specify which search engines to use

### PR DESCRIPTION
**Pull request by Hainish**
This is useful for https://github.com/EFForg/https-everywhere, see issue https://github.com/EFForg/https-everywhere/issues/7277#issuecomment-251787466

With this, we can specify what search engines to use - best configuration in our case is

    python sublist3r.py -d example.com -e Baidu -e Yahoo -e Google -e Bing -e Ask -e Netcraft -e Virustotal -e "SSL Certificates"

### CODE REVIEW

**Comment by Hainish** [comment]
In the default configuration, all engines are used.

**Comment by Hainish** [comment]
@the-storm: I definitely get that.  The reason why I considered it acceptable in this case is because the conditionals effectively whitelist what will ever pass through to the `eval` statement.  I can open another PR following your suggestion if you like, though.

**Comment by the-st0rm** [comment]
@Hainish  I think that would be very good. So I can merge the PR right away :)

**Comment by Hainish** [comment]
@the-st0rm: done

**Comment by aboul3la** [comment]
Hello @Hainish,

Thanks for your PR. We're glad to support EFF and https-everywhere.

I just have some few suggestions to improve this PR before I can merge it.

1. I don't think it's good to use many if conditions hard-coded inside the code. I think the optimal way is to use a dict that stores all the possible engines and match the user input against it.

2. it's better to use a comma separated list for the engine argument rather than using multiple -e args  
such as``` -e yahoo,google,bing,baidu```  it would be better than doing ``` -e yahoo -e google -e bing -e baidu ```

3. It's better to use case insensitive matching for the engines names. As your current code will only work if the argument is like ``` -e Yahoo```  but it won't work if it's  ```-e yahoo``` 

Below is a little code snippet i wrote on how the code should be:

```python

    supported_engines = {'baidu':BaiduEnum,
                         'yahoo':YahooEnum,
                         'google':GoogleEnum,
                         'bing':BingEnum,
                         'ask':AskEnum,
                         'netcraft':NetcraftEnum,
                         'dnsdumpster':DNSdumpster,
                         'virstotal':Virustotal,
                         'threatcrowd':ThreatCrowd,
                         'ssl':CrtSearch,
                         'passivedns':PassiveDNS
                         }

    engines = engines.split(',')
    chosenEnums = []

    if engines == None:
        chosenEnums = (BaiduEnum, YahooEnum, GoogleEnum, BingEnum, AskEnum,
                       NetcraftEnum, DNSdumpster, Virustotal, ThreatCrowd, CrtSearch, PassiveDNS)
    else:
        for engine in engines:
            if supported_engines.has_key(engine.lower()):
                chosenEnums.append(supported_engines[engine.lower()])

Kindly update the PR so I can merge it.
Thanks

**Comment by Hainish** [comment]
@aboul3la: Thanks for the words of support! I've updated this PR.

**Comment by aboul3la** [comment]
Thank you @Hainish. It's merged now

File: sublist3r.py [hunk 1]
```diff
@@ -915,8 +916,32 @@ def main(domain, threads, savefile, ports, silent, verbose, enable_bruteforce):
...
+
+    chosenEnums = []
+
+    if engines == None:
+        chosenEnums = [BaiduEnum, YahooEnum, GoogleEnum, BingEnum, AskEnum,
+                       NetcraftEnum, DNSdumpster, Virustotal, ThreatCrowd, CrtSearch, PassiveDNS]
+    else:
+        engines = engines.split(',')
+        for engine in engines:
+            if supported_engines.has_key(engine.lower()):
```
Comment (Foorack) [inline sublist3r.py]: @aboul3la This breaks Python3 compatibility because 'has_key' function was removed in Python 3.

**Pull request merged**